### PR TITLE
Pizza Overhaul 2022

### PR DESCRIPTION
**Pull request by PacifistDalek**
## What Does This PR Do
- ports pizza, pizza box, and pizza bomb sprites from TG (with a bunch of tweaks by yours truly)
- gives pizza bombs an active sprite (animated flashing lights to make it even more intense!)
- ~~changes how pizzas are made, before you threw the components in the oven and that made a pizza. You now use the crafting menu to create a raw pizza, and you place that in the oven to make a cooked pizza. Raw pizzas are still edible if you don't have an oven, but give half nutrients and aren't sliceable.~~ **Removed because of #18352 as discussed below. Pizzas still are cooked by putting the ingredients in the oven.
- tweaks some recipes to make more sense either for their sprites or for how they're made in real life
- removes the syntipizza recipe which just duplicated the meat pizza and didn't even give it the name syntipizza
- adds **NEW PIZZAS**: Cheese, Pepperoni, Donk-Pocket, Dank, Garlic, Firecracker, and "Pesto". (I didn't think it was worth adding two new growables and everything that came with it just to create TG's realistic pesto pizza recipe, so instead I went with comedy and created a wasabi pizza, thus giving wasabi an extra food use as well)

## Why It's Good For The Game
pizza-time

## Images of changes
(Raw pizza sprite removed, see above)
pizza1
image

## Changelog
:cl:
add: Seven new pizzas!
tweak: some pizza recipes are tweaked slightly
imageadd: New pizza, pizza box, and pizza bomb sprites

/:cl:

### CODE REVIEW

**Comment by Adrer** [comment]
Okay, but real talk, donk-pockets are pizza turnovers, how do you make a pizza turnover pizza.

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm [hunk 1]
```diff
@@ -287,3 +287,159 @@
...
+	name = "Uncooked Garlic Pizza"
+	reqs = list(
+		/obj/item/reagent_containers/food/snacks/sliceable/flatdough = 1,
+		/obj/item/reagent_containers/food/snacks/grown/garlic = 2,
+		/obj/item/reagent_containers/food/snacks/cheesewedge = 1,
+		/datum/reagent/consumable/garlic = 5,
+	)
+	result = list(/obj/item/reagent_containers/food/snacks/rawpizza/garlicpizza)
+	category = CAT_FOOD
+	subcategory = CAT_PIZZA
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm]:
```suggestion
	subcategory = CAT_PIZZA

```
new line

File: code/modules/food_and_drinks/recipes/recipes_oven.dm [hunk 2]
```diff
@@ -215,84 +215,87 @@
...
+	items = list(
+		/obj/item/reagent_containers/food/snacks/rawpizza/pestopizza,
+	)
+	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/pestopizza
+
+/datum/recipe/oven/garlicpizza
+	items = list(
+		/obj/item/reagent_containers/food/snacks/rawpizza/garlicpizza,
+	)
+	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/garlicpizza
```
Comment (Sirryan2002) [inline code/modules/food_and_drinks/recipes/recipes_oven.dm:295]:
```suggestion
/datum/recipe/oven/cheesepizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/cheesepizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/cheesepizza

/datum/recipe/oven/pepperonipizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/pepperonipizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/pepperonipizza

/datum/recipe/oven/donkpocketpizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/donkpocketpizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/donkpocketpizza

/datum/recipe/oven/dankpizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/dankpizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/dankpizza

/datum/recipe/oven/firecrackerpizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/firecrackerpizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/firecrackerpizza

/datum/recipe/oven/pestopizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/pestopizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/pestopizza

/datum/recipe/oven/garlicpizza
	items = list(/obj/item/reagent_containers/food/snacks/rawpizza/garlicpizza)
	result = /obj/item/reagent_containers/food/snacks/sliceable/pizza/garlicpizza
```

File: code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm (hunk 1 above)
Comment (Sirryan2002) [reply code/modules/food_and_drinks/recipes/tablecraft/recipes_table.dm]: oh I think you fixed this aleady

**Comment by Sirryan2002** [comment]
Also you're failing CL because you changed some pizza type paths but didn't edit the maps they were in

**Comment by KalevTait** [comment]
While you are in pizza_bomb, do you want to change `timer = (input(user, "Set a timer, from one second to ten seconds.", "Timer", "[timer]") as num) * 10` to `timer = (input(user, "Set a timer, from one second to ten seconds.", "Timer", "[timer]") as num) SECONDS`?

**Comment by Bm0n** [comment]
Why didn't you port TGs new pizza box inhands? A little weird that you'd port the new box sprite but not is inhands

**Comment by PacifistDalek** [comment]
A whoops, the map issue and the in-hands were my bad.

@KalevTait will do!

Thanks @Sirryan2002 for the suggested changes and for catching what was failing CL

AND I've been informed in discord that crafting menu recipes don't carry forward the reagents from their ingredients the way they should, so I should possibly switch back to just throwing ingredients in the oven to make the pizzas.

**Comment by PacifistDalek** [comment]
Made it a draft because i have to figure out what to do with the new pizza cooking method.

**Comment by Bm0n** [comment]
you probably should have updated your branch before adding the new inhands, now a bunch of new weapon inhands from #18118 have been deleted

**Comment by PacifistDalek** [comment]
Yes, I have merge conflicts to fix, but i've re-opened it from draft state because beyond that I think it should be ready to go.

**Review by Sirryan2002 — APPROVED** [review]
code looks good

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/food_and_drinks/food/foods/pizza.dm [hunk 3]
```diff
@@ -252,8 +374,9 @@
 		return
 	..()
 
+
 /obj/item/pizzabox/margherita/Initialize(mapload)
-	pizza = new /obj/item/reagent_containers/food/snacks/sliceable/pizza/margherita(src)
+	pizza = new /obj/item/reagent_containers/food/snacks/sliceable/pizza/margheritapizza(src)
 	box_tag = "margherita deluxe"
 	return ..()
```
Comment (AffectedArc07) [inline code/modules/food_and_drinks/food/foods/pizza.dm:381]: Initialize has the parent call first, not last.

Please convert
```dm
/obj/item/pizzabox/margherita/Initialize(mapload)
	pizza = new /obj/item/reagent_containers/food/snacks/sliceable/pizza/margheritapizza(src)
	box_tag = "margherita deluxe"
	return ..()
```

to
```dm
/obj/item/pizzabox/margherita/Initialize(mapload)
	. = ..()
	pizza = new /obj/item/reagent_containers/food/snacks/sliceable/pizza/margheritapizza(src)
	box_tag = "margherita deluxe"
```

And do the same for the others.

**Review by AffectedArc07 — CHANGES_REQUESTED** [review]

**Review by SteelSlayer — APPROVED** [review]

**Review by hal9000PR — APPROVED** [review]

**Pull request merged**